packages/user_packages/iotcv2/iotcv2_toyota_sim.py

```diff
@@ -0,0 +1,98 @@
+from liota.dccs.iotcv2 import Iotcv2
+from liota.lib.utilities.utility import read_user_config
+from liota.dccs.dcc import RegistrationFailure
+from liota.core.package_manager import LiotaPackage
+from liota.lib.utilities.identity import Identity
+from liota.dcc_comms.mqtt_dcc_comms import MqttDccComms
+from liota.lib.utilities.tls_conf import TLSConf
+import logging
+import os
+
+log = logging.getLogger(__name__)
+
+dependencies = []
+
+class PackageClass(LiotaPackage):
+    """
+    This package creates a IoTCV2 DCC object, registers Toyota edge system (TES),
+    sets its properties, register 19 ECUs, sets theirs properties, and creates
+    relationships between TES and ECUs on IoTCV2
+    """
+
+    def run(self, registry):
+        from liota.entities.edge_systems.general_edge_system import GeneralEdgeSystem
+        from liota.entities.devices.simulated_device import SimulatedDevice
+
+        # Get values from configuration file
+        config_path = registry.get("package_conf")
+        self.config = read_user_config(config_path + '/user_packages/iotcv2/sampleProp.conf')
+
+        # Initialize edgesystem
+        edge_system = GeneralEdgeSystem(os.environ['VIN'])
+        registry.register("edge_system", edge_system)
+
+        # Encapsulate Identity
+        identity = Identity(root_ca_cert=self.config['broker_root_ca_cert'],
+                            username=self.config['broker_username'],
+                            password=self.config['broker_password'],
+                            cert_file=None, #self.config['edge_system_cert_file'],
+                            key_file=None #self.config['edge_system_key_file']
+                           )
+
+        tls_conf = TLSConf(self.config['cert_required'],
+                           self.config['tls_version'], None)
+
+        # Initialize DCC object with MQTT transport
+        self.iotcc = Iotcv2(
+                                      MqttDccComms(edge_system_name=edge_system.name,
+                                                   url=self.config['BrokerIP'],
+                                                   port=self.config['BrokerPort'],
+                                                   identity=identity,
+                                                   tls_conf=tls_conf,
+                                                   enable_authentication=True
+                                                   )
+                                      )
+        try:
+            self.iotcc_edge_system = self.iotcc.register(edge_system)
+            registry.register("iotc_v2_mqtt_wssl", self.iotcc)
+            registry.register("iotc_v2_mqtt_wssl_edge_system", self.iotcc_edge_system)
+        except RegistrationFailure:
+            print "EdgeSystem registration to IOTCV2 failed"
+            return
+        self.iotcc.set_properties(self.iotcc_edge_system, self.config['EdgeSystemPropList'])
```

Comment:
We are not catching the Exceptions in set_properties.

Reply:
Should we catch an exception for this set_properties call?

Reply:
already added.